cpdb/cpdb-frontend.c

```diff
@@ -492,6 +394,152 @@ cpdb_printer_obj_t *cpdbGetDefaultPrinter(cpdb_frontend_obj_t *f)
     return default_printer;
 }
 
+int cpdbMakeUserDefault(cpdb_printer_obj_t *p)
+{
```

Comment:
`cpdbMakeUserDefault()` and `cpdbMakeSystemDefault()` create the same type of default printer file but only differ in the file mane under which they store the file. Here you could create one function for reading the file, adding the new entry and removing the duplicates, taking the file name and the printer object as parameters. Then you create two small wrapper functions `cpdbSetUserDefault()` and `cpdbSetSystemDefault()` calling your new function with the `CPDB_USER_DEFAULT_PRINTERS` and `CPDB_SYSTEM_DEFAULT_PRINTERS` resp. as first parameter and `p` as second parameter.